src/tests/CMakeLists.txt

```diff
@@ -1,10 +1,24 @@
+option(USE_SYSTEM_CATCH2 "Use system Catch2 v3 otherwise build it locally" ON)
+
+if(USE_SYSTEM_CATCH2)
+  find_package(Catch2 3 REQUIRED)
+else()
+  include(FetchContent)
+  FetchContent_Declare(
+    Catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+    GIT_TAG        v3.3.2
+  )
+  FetchContent_MakeAvailable(Catch2)
+  list(APPEND CMAKE_MODULE_PATH ${Catch2_SOURCE_DIR}/extras)
+endif()
 
-# Check if catch2 exists and unit tests can be built
-find_package(Catch2 3)
 if(Catch2_FOUND)
-    add_subdirectory(algorithms_test)
+  include(CTest)
```

Comment:
If I'm not mistaken, calling `enable_testing` would work best from the top level of the project. Doing in a subdirectory would require more directory changes: `cd build && make && cd src/tests && ctest`.

Reply:
Agreed. This doesn't work yet in the way I would like...